app/coffee/ResourceStateManager.coffee

```diff
@@ -33,14 +34,30 @@ module.exports = ResourceStateManager =
 					return callback()
 		else
 			logger.log state:state, basePath:basePath, "writing sync state"
-			fs.writeFile stateFile, state, {encoding: 'ascii'}, callback
+			resourceList = (resource.path for resource in resources)
+			fs.writeFile stateFile, [resourceList..., "stateHash:#{state}"].join("\n"), callback
 
-	checkProjectStateHashMatches: (state, basePath, callback) ->
+	checkProjectStateHashMatches: (state, basePath, callback = (error, resources) ->) ->
 		stateFile = Path.join(basePath, @SYNC_STATE_FILE)
-		SafeReader.readFile stateFile, 64, 'ascii', (err, oldState) ->
+		SafeReader.readFile stateFile, 128*1024, 'utf8', (err, result) ->
 			return callback(err) if err?
-			logger.log state:state, oldState: oldState, basePath:basePath, stateMatches: !(state isnt oldState), "checking sync state"
-			if state isnt oldState
+			[resourceList..., oldState] = result?.toString()?.split("\n") or []
+			newState = "stateHash:#{state}"
+			logger.log state:state, oldState: oldState, basePath:basePath, stateMatches: (newState is oldState), "checking sync state"
+			if newState isnt oldState
 				return callback new Errors.FilesOutOfSyncError("invalid state for incremental update")
 			else
-				callback(null)
+				resources = ({path: path} for path in resourceList)
+				callback(null, resources)
+
+	checkResourceFiles: (resources, allFiles, directory, callback = (error) ->) ->
+		# check if any of the input files are not present in list of files
```

Comment:
This seems like it should never happen - haven't we just written all of the input files to disk? I'm not sure I understand the failure mode this is checking against.

Reply:
This is just for the incremental case.  We do this check immediately before the incremental files are updated, they should be present on disk from the last compilation in this case.  If not, something has gone out of sync - hence the check.

Reply:
👍